packages/react/src/components/tooltip.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { FunctionComponent } from "react";
+import * as TooltipPrimitive from "@radix-ui/react-tooltip";
+import type { TooltipContentProps } from "@radix-ui/react-tooltip";
+import { clx } from "../utils";
+
+const TooltipProvider = TooltipPrimitive.TooltipProvider;
+const TooltipRoot = TooltipPrimitive.Root;
+const TooltipTrigger = TooltipPrimitive.Trigger;
+const TooltipPortal = TooltipPrimitive.Portal;
+
+const TooltipContent = React.forwardRef<
+  React.ElementRef<typeof TooltipPrimitive.Content>,
+  React.ComponentPropsWithoutRef<typeof TooltipPrimitive.Content>
+>(({ className, asChild, children, ...props }, ref) => (
+  <TooltipPrimitive.Content
+    ref={ref}
+    side="top"
+    align="start"
+    asChild={false}
+    className={clx(
+      "bg-bg-black-900 border-bg-black-900 shadow-context-menu text-txt-white data-[state=delayed-open]:data-[side=bottom]:animate-slideUpAndFade data-[state=delayed-open]:data-[side=left]:animate-slideRightAndFade data-[state=delayed-open]:data-[side=right]:animate-slideLeftAndFade data-[state=delayed-open]:data-[side=top]:animate-slideDownAndFade z-50 min-w-[50px] max-w-[250px] rounded-sm border px-3 py-2 will-change-[transform,opacity]",
+      className,
+    )}
+    {...props}
+  >
+    {children}
+  </TooltipPrimitive.Content>
+));
+
+TooltipContent.displayName = TooltipPrimitive.Content.displayName;
+
+const TooltipArrow = React.forwardRef<
+  React.ElementRef<typeof TooltipPrimitive.Arrow>,
+  React.ComponentPropsWithoutRef<typeof TooltipPrimitive.Arrow>
+>(({ className, ...props }, ref) => (
+  <TooltipPrimitive.Arrow
+    className={clx("fill-bg-black-900", className)}
+    {...props}
+  />
+));
+
+/**
+ * Props for Tooltip component.
+ * @typedef TooltipProps
+ */
+
+interface TooltipProps
+  extends React.ComponentPropsWithoutRef<typeof TooltipPrimitive.Root>,
+    Omit<TooltipContentProps, "content"> {
+  content: React.ReactNode;
+  container?: React.ComponentPropsWithoutRef<
+    typeof TooltipPrimitive.Portal
+  >["container"];
+  forceMount?: true;
+  className?: string;
+}
+
+/**
+ * Tooltip component.
+ */
+const Tooltip: FunctionComponent<TooltipProps> = ({
+  container,
+  defaultOpen,
+  delayDuration,
+  disableHoverableContent,
+  onOpenChange,
+  open,
+  content,
+  children,
+  className,
+  forceMount,
+  ...contentProps
+}) => {
+  const rootProps = {
+    open,
+    defaultOpen,
+    onOpenChange,
+    delayDuration,
+    disableHoverableContent,
+  };
+
+  // Check if the classname prop passed contain `bg-*` class, then return the color applied.
+  const getCssBgColor = (property: "fill" | "border", className?: string) => {
+    if (!className) {
+      return null;
+    }
+    const bgClassRegex = /\bbg-([a-zA-Z0-9-_]+)\b/;
+    const match = className.match(bgClassRegex);
+
+    return match ? `${property}-${match[1]}` : null;
+  };
```

Comment:
No, this wont work & can be a potential bug.

_Issue_
Tailwind will not be able to detect this. When tailwind compiles, it checks for all the CSS class that is used and strips whichever that is not used. Since, this is dynamically composing the CSS class with JS or template string, this wont work.

Reply:
Cant the fill color be `inherit` or something?

Reply:
no `inherit` wont work because when using the `Arrow` primitives, its is wrapped in a `<span />` tag that cannot be tweaked and `Arrow` is referring to the svg wrapped within the span.

```
// This is the Content primitives
<div data-side="bottom" data-align="center" data-state="instant-open" class="border-bg-black-900 shadow-context-menu text-txt-white data-[state=delayed-open]:data-[side=bottom]:animate-slideUpAndFade data-[state=delayed-open]:data-[side=left]:animate-slideRightAndFade data-[state=delayed-open]:data-[side=right]:animate-slideLeftAndFade data-[state=delayed-open]:data-[side=top]:animate-slideDownAndFade fill-bg-danger-600 z-50 min-w-[50px] max-w-[250px] rounded-sm border px-3 py-2 will-change-[transform,opacity] bg-bg-danger-600" style="--radix-tooltip-content-transform-origin: var(--radix-popper-transform-origin); --radix-tooltip-content-available-width: var(--radix-popper-available-width); --radix-tooltip-content-available-height: var(--radix-popper-available-height); --radix-tooltip-trigger-width: var(--radix-popper-anchor-width); --radix-tooltip-trigger-height: var(--radix-popper-anchor-height);">

  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.


  // the span tag here is wrapping the Arrow and used by Radix to position the arrow. cant find ways to tweak this tag
  <span style="position: absolute; transform: rotate(180deg); left: 232px; top: 0px; transform-origin: center 0px;">
    // This is the Arrow. passing className to `Arrow` is reflected here
    <svg class="fill-current" width="10" height="5" viewBox="0 0 30 10" preserveAspectRatio="none" style="display: block;"><polygon points="0,0 30,0 15,10"></polygon></svg>
  </span>

    <span id="radix-:Rq9ukvajt6:" role="tooltip" style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; overflow-wrap: normal;">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</span>

</div>
```

Reply:
but you are right, the function wont work here as u mentioned.

Reply:
Actually, we dont need to support this. Just realised this is meant to make sure the arrow bg is the same as custom bg. We only need to support the light, dark color in myds

Reply:
not just custom bg per say. if user pass myds color for example `bg-bg-danger-600` to `Content` className prop, the background of content will change the color but not the arrow. and the color is still myds color scheme. 

and ideally, i feel like i wouldnt want the user to have to declare the color twice to Content and Arrow just to get the color to be in sync.

Reply:
unless you want me to remove the className prop.